src/PointwiseFunctions/AnalyticSolutions/Poisson/Lorentzian.hpp

```diff
@@ -0,0 +1,75 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/Prefixes.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"     // IWYU pragma: keep
+#include "Elliptic/Systems/Poisson/Tags.hpp"    // IWYU pragma: keep
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+class DataVector;
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace Poisson {
+namespace Solutions {
+
+/*!
+ * \brief A Lorentzian solution to the Poisson equation
+ *
+ * \details This implements the Lorentzian solution
+ * \f$u(\boldsymbol{x})=\left(1+r^2\right)^{-\frac{1}{2}}\f$ to the
+ * three-dimensional Poisson equation
+ * \f$-\Delta u(\boldsymbol{x})=f(\boldsymbol{x})\f$, where
+ * \f$r^2=x^2+y^2+z^2\f$. The corresponding source is
+ * \f$f(\boldsymbol{x})=3\left(1+r^2\right)^{-\frac{5}{2}}\f$.
+ *
+ * \note Corresponding 1D and 2D solutions are not implemented yet.
+ */
+template <size_t Dim>
+class Lorentzian {
+ public:
+  using options = tmpl::list<>;
+  static constexpr OptionString help{
+      "A Lorentzian solution to the Poisson equation."};
+
+  Lorentzian() = default;
+  Lorentzian(const Lorentzian&) noexcept = delete;
+  Lorentzian& operator=(const Lorentzian&) noexcept = delete;
+  Lorentzian(Lorentzian&&) noexcept = default;
+  Lorentzian& operator=(Lorentzian&&) noexcept = default;
+  ~Lorentzian() noexcept = default;
+
+  auto field_variables(const tnsr::I<DataVector, Dim>& x) const noexcept
+      -> tuples::TaggedTuple<Field>;
+
+  auto source_variables(const tnsr::I<DataVector, Dim>& x) const noexcept
+      -> tuples::TaggedTuple<::Tags::Source<Field>,
+                             ::Tags::Source<AuxiliaryField<Dim>>>;
+
+  // clang-tidy: no pass by reference
+  void pup(PUP::er& p) noexcept;  // NOLINT
+};
+
+template <size_t Dim>
+constexpr bool operator==(const Lorentzian<Dim>& /*lhs*/,
```

Comment:
Move to cpp.

Reply:
I though with the `constexpr` this is would be inlined, so it would trivially evaluate to `true`, which would be better than moving this to the cpp. Am I wrong, and if so, why? I made the change in the fixup comments, just would like to understand what's going on.

Reply:
Fore rarely called functions (this one will probably never be called outside the test) we would prefer to take the slight runtime performance hit to get the slight compile-time performance improvement from not having to parse it when the hpp is included.